kubeflow/trainer/rhai/traininghub_instrumentation.py

```
@@ -0,0 +1,961 @@
+                        first_line = f.readline().strip()
+                        if first_line:
+                            config = json.loads(first_line)
+                except Exception:
+                    print(
+                        "[Kubeflow] Warning: Failed to read SFT config",
+                        flush=True,
+                    )
+
+                # Read last line for latest metrics using tail
+                try:
+                    result = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)